src/phpDocumentor/Descriptor/Builder/PhpParser/ClassAssembler.php

```diff
@@ -31,22 +31,18 @@ class ClassAssembler extends AssemblerAbstract
      */
     public function create($data)
     {
-        $classDescriptor = new ClassDescriptor();
-
-        $this->assembleDocBlock($data->docBlock, $classDescriptor);
-
-        $classDescriptor->setFullyQualifiedStructuralElementName('\\' . $data->namespacedName->toString());
-        $classDescriptor->setName($data->name);
-        $classDescriptor->setLine($data->getLine());
-        $classDescriptor->setParent($data->extends ? '\\' . $data->extends->toString() : null);
-        $classDescriptor->setAbstract($data->isAbstract());
-        $classDescriptor->setFinal($data->isFinal());
-
-        $classDescriptor->setNamespace('\\' . $this->extractNamespace($data));
+        $classDescriptor = new ClassDescriptor(
+            new Fqsen('\\' . $data->namespacedName->toString()),
+            $data->docBlock,
+            $data->extends ? new ClassDescriptor(new Fqsen('\\' . $data->extends)) : null,
+            $data->isAbstract(),
+            $data->isFinal()
+        );
 
         foreach ($data->implements as $interfaceClassName) {
-            $interfaceFqcn = '\\' . $interfaceClassName->toString();
-            $classDescriptor->getInterfaces()->set($interfaceFqcn, $interfaceFqcn);
+            $classDescriptor->addInterface(
+                new Interface_(new Fqsen($interfaceFqcn = '\\' . $interfaceClassName->toString()))
```

Comment:
This should just be the Fqsen of the interface because it is only a relation between this class and an interface; the actual interface element is built once it is encounted in the Files